xskillscore/core/probabilistic.py

```diff
@@ -526,73 +622,153 @@ def rps(
     Returns
     -------
     xarray.Dataset or xarray.DataArray:
-        ranked probability score
+        ranked probability score with coords ``forecasts_category_edge`` and ``observations_category_edge`` as str
+
 
     Examples
     --------
-    >>> observations = xr.DataArray(np.random.normal(size=(3,3)),
+    >>> observations = xr.DataArray(np.random.random(size=(3,3)),
     ...                             coords=[('x', np.arange(3)),
     ...                                     ('y', np.arange(3))])
-    >>> forecasts = xr.DataArray(np.random.normal(size=(3,3,3)),
+    >>> forecasts = xr.DataArray(np.random.random(size=(3,3,3)),
     ...                          coords=[('x', np.arange(3)),
     ...                                  ('y', np.arange(3)),
     ...                                  ('member', np.arange(3))])
-    >>> category_edges = np.array([.2, .5, .8])
-    >>> rps(observations > 0.5, (forecasts > 0.5).mean('member'), category_edges)
-    <xarray.DataArray 'histogram_category' (y: 3)>
-    array([1.        , 1.        , 0.33333333])
+    >>> category_edges = np.array([.0, .5, 1.])
+    >>> xs.rps(observations, forecasts, category_edges, dim='x')
+    <xarray.DataArray (y: 3)>
+    array([0.85185185, 0.59259259, 0.37037037])
     Coordinates:
-      * y        (y) int64 0 1 2
+      * y                           (y) int64 0 1 2
+        forecasts_category_edge     <U56 '[[0.0, 0.33), [0.33, 0.66)), [[0.33, 0....
+        observations_category_edge  <U56 '[[0.0, 0.33), [0.33, 0.66)), [[0.33, 0....
+
+    You can also define multi-dimensional ``category_edges``, e.g. with xr.quantile.
+    However, you still need to ensure that ``category_edges`` covers the forecasts and
+    observations distributions.
+    >>> category_edges = xr.concat([
+    ...     xr.DataArray(0).expand_dims('category_edge').assign_coords(category_edge=[0]),
+    ...     observations.quantile(q=[.33, .66]).rename({'quantile':'category_edge'}),
+    ...     xr.DataArray(1).expand_dims('category_edge').assign_coords(category_edge=[1])
+    ... ],'category_edge')
+    >>> xs.rps(observations, forecasts, category_edges, dim='x')
+    <xarray.DataArray (y: 3)>
+    array([1.18518519, 0.85185185, 0.40740741])
+    Coordinates:
+      * y                           (y) int64 0 1 2
+        forecasts_category_edge     <U56 '[[0.0, 0.33), [0.33, 0.66)), [[0.33, 0....
+        observations_category_edge  <U56 '[[0.0, 0.33), [0.33, 0.66)), [[0.33, 0....
```

Comment:
@judithberner @abjaye do this look like suitable API for you? The definition of category_edges is a bit tricky but I have a convenience function `quantile_edges_pad_lower_upper` for that also.